libkbfs/folder_branch_ops.go

```diff
@@ -1176,6 +1176,16 @@ func (fbo *folderBranchOps) SetInitialHeadFromServer(
 		return fmt.Errorf("MD with revision=%d not initialized", md.Revision())
 	}
 
+	// Return early if the head is already set.  This avoids taking
```

Comment:
That's right, but it's only an issue for the libkbfs DSL engine -- everything else properly caches the root node as far as I know.  I'll add a TODO in the engine instead.